virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -97,6 +97,39 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
         .filter(win => win.windowState.stateType === DesktopWindowStateType.Maximized)
         .forEach(win => this.refreshMaximizedWindowSize(win));
     });
+
+    let tabHandler = (event: KeyboardEvent) => {
+      if(this.focusedWindow != null){
+        if(event.keyCode == 9 || event.which == 9){ // tab
+          let activeViewportID = this.getViewportIdFromDOM(document.activeElement);
+          if(activeViewportID != Number(this.focusedWindow.viewportId)){
+            let focusHTML = this.getHTML(this.focusedWindow.windowId);
+            let focusTabIndex = focusHTML.getAttribute('tabindex');
+            focusHTML.setAttribute('tabindex', '0');
+            focusHTML.focus();
+            focusHTML.setAttribute('tabindex', focusTabIndex);
+          }
+        }
+      }
+    }
+    document.addEventListener('keyup', tabHandler, false);
+    document.addEventListener('keydown', tabHandler, false);
+  }
+
+  private getViewportIdFromDOM(element: any): Number{
+    var parentViewportElement: any;
+    var head = element;
+    while(head.parentNode !== document){
+      if(head.parentNode.nodeName.toLowerCase() === 'com-rs-mvd-viewport'){
```

Comment:
Just adding comments from what we discussed for sake of memory:

'com-rs-mvd-viewport' should be a variable name, in case we change it down the road.